tests/operators/test_autodiff.py

```diff
@@ -81,22 +91,38 @@ def test_adjoint_and_grad(operator, ndft_matrix, interface):
 
 @pytest.mark.parametrize("interface", ["torch-gpu", "torch-cpu"])
 @pytest.mark.skipif(not TORCH_AVAILABLE, reason="Pytorch is not installed")
-def test_forward_and_grad(operator, ndft_matrix, interface):
+def test_forward_and_grad(operator, interface):
     """Test the adjoint and gradient of the operator."""
     if operator.backend == "finufft" and "gpu" in interface:
         pytest.skip("GPU not supported for finufft backend")
 
-    ndft_matrix_torch = to_interface(ndft_matrix, interface=interface)
+    if torch.is_tensor(operator.samples):
+        operator.samples = operator.samples.cpu().detach().numpy()
+
+    operator.samples = to_interface(operator.samples, interface=interface)
```

Comment:
Following on to my previous comment. Something to think about, this makes using with autodiff slightly complicated as users cannot just do:

trajectory = torch.Tensor(...)
trajectory.requires_grad = True

get_operator('cufinufft', grad_wrt_traj=True)(trajectory)


and then get gradients. Right? I think we ought to support that. Need not be in this PR. Maybe when we are getting example in place in #104 , we better have a simpler version of codes. @alineyyy @paquiteau What do you think?

Reply:
Yes. We will track this in #127